File: python/duplicates/217.contains_duplicates.py
# leetcode Problem 217

# using set

def containsDuplicates(nums: List(int)) -> bool:
	s = set()
	for num in nums:
		if num in s:
			return True
		s.add(num)
	return False


# Runtime: 19ms | Beats 32.59%
# Memory: 31.70 MB | Beats 29.59%

File: cpp/duplicates/217.Contains_Duplicates.cpp
// leetcode Problem 217

#include<iostream>
#include<vector>
#include<unordered_map>
using namespace std;

// using unordered_map

bool hasDuplicates(vector<int>& nums){
	unordered_map<int, int> s;
	for(int num : nums){
		if(s.find(num) != s.end()){
			return true;
		}
		s[nums] = 1;
	}
	return false;
}


// Runtime: 48ms | Beats: 62.69%
// Memory: 90.95 MB | 30.90%